File: pdui-express/lib/core/PDUIPopNavigation.ts
import { PDUIFn } from "../models/PDUIFn";
import type {
    PBFnExpression,
    PBPopNavigation,
} from "../proto-out/pdui-proto-out";

export class PDUIPopNavigation extends PDUIFn {
    constructor() {
        super({
            identity: "popNavigation",
            name: undefined,
            params: {},
            body: {},
        });
    }

    //TODO: Add return data
    override toPB(): PBPopNavigation {
        return {};
    }

    //TODO: Add return data
    override toFnExpression(): PBFnExpression {
        return {
            identity: "popNavigation",
            payload: { oneofKind: "popNavigation", popNavigation: {} },
        };
    }
}

File: pdui_flutter/lib/core/pdui_pop_navigation.dart
import 'package:flutter/material.dart';
import 'package:pdui_flutter/core/pdui_context.dart';
import 'package:pdui_flutter/proto_out/pdui-proto-out.pb.dart';

import '../models/pdui_fn.dart';

class PduiPopNavigation extends PduiFn {
  PBPopNavigation pbPopNavigation;

  PduiPopNavigation(this.pbPopNavigation) : super("popNavigation");

  @override
  resolve() {
    Navigator.of(PduiContext.navigatorKey.currentContext!).pop();
  }
}
